### DISCUSSION TITLE
Aalen Johansen Not Implemented Dev Status

### DISCUSSION THREAD

**Discussion by mike-dobbles** [CamDavidsonPilon/lifelines#1361] [Q&A] [answered]
I noticed a number of Aalen Johansen estimator methods give not implemented errors including:
cumulative_density_at_times(times, label=None)
cumulative_hazard_at_times(times, label=None)

I've worked around the missing cumulative_density_at_times method and am getting direct access to the calculation results via the private properties of timeline, cumulative_density_["CIF_2"],  confidence_interval_cumulative_density_["AJ_estimate_lower_0.95"], and confidence_interval_cumulative_density_["AJ_estimate_upper_0.95"]

I don't immediately see a way to get to a hazard property.  Any suggestions?

**Comment by CamDavidsonPilon** [comment, marked as answer]
Hi @mike-dobbles, the hazard isn't directly estimated, so you'd have to compute it yourself (using the cumulative_hazard)
